unison-cli/src/Unison/Codebase/Editor/HandleInput.hs

```diff
@@ -2223,20 +2229,33 @@ resolveConfiguredUrl ::
 resolveConfiguredUrl pushPull destPath' = ExceptT do
   currentPath' <- use LoopState.currentPath
   let destPath = Path.resolve currentPath' destPath'
-  let configKey = gitUrlKey destPath
-  (eval . ConfigLookup) configKey >>= \case
-    Just url ->
-      case P.parse UriParser.writeRemotePath (Text.unpack configKey) url of
+  let remoteMappingConfigKey = remoteMappingKey destPath
+  (eval . ConfigLookup) remoteMappingConfigKey >>= \case
+    Nothing -> do
+      let gitUrlConfigKey = gitUrlKey destPath
+      -- Fall back to deprecated GitUrl key
+      (eval . ConfigLookup) gitUrlConfigKey >>= \case
+        Just url ->
+          case WriteRemotePathGit <$> P.parse UriParser.deprecatedWriteGitRemotePath (Text.unpack gitUrlConfigKey) url of
+            Left e ->
+              pure . Left $
+                ConfiguredRemoteMappingParseError pushPull destPath url (show e)
+            Right ns ->
+              pure . Right $ ns
+        Nothing ->
+          pure . Left $ NoConfiguredRemoteMapping pushPull destPath
+    Just url -> do
+      case P.parse UriParser.writeRemotePath (Text.unpack remoteMappingConfigKey) url of
         Left e ->
           pure . Left $
-            ConfiguredGitUrlParseError pushPull destPath' url (show e)
+            ConfiguredRemoteMappingParseError pushPull destPath url (show e)
```

Comment:
It looks like you could have an error in your `GitUrl` section and UCM would tell you you have an error in your `RemoteMapping` section; I think we should have separate errors for each, even though it's more typing. (Or it could be parameterized.)